src/js/common/lib/i18n.js

```diff
@@ -0,0 +1,188 @@
+/*
+
+    Functions to assist in internationalising and localising content.
+
+*/
+/* eslint-env node */
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.handlebars.i18n");
+
+/**
+ *
+ * Parse the content of an `Accept-Language` header and return the full list of locales in order of preference.
+ *
+ * @param {String} header - The value of the `Accept-Language` header.
+ * @return {String[]} - An array of strings representing locales and/or languages, in order of preference.
+ *
+ */
+gpii.handlebars.i18n.getAllLocalesFromHeader = function (header) {
+    var acceptedLanguages = [];
+    if (header === undefined) {
+        acceptedLanguages.push("*");
+    }
+    else if (typeof header === "string") {
+        var localeSegments = header.trim().split(/ *, */);
+        var segmentsAsObjects = [];
+        fluid.each(localeSegments, function (localeSegment) {
```

Comment:
We tend to just use Array.forEach for known definite arrays now since it is available everywhere and there are clear performance advantages

Reply:
Done.